src/client/workflow/workflow.ts

```diff
@@ -125,11 +125,50 @@ export class Workflow<TInitialRequest = unknown> {
     // add result to pending and send request
     this.addResult(result);
     await this.sendPendingToQstash();
-    this.skip = true;
 
     return result;
   }
 
+  /**
+   *
+   * @param stepName
+   * @param duration sleep duration in seconds
+   */
+  public async sleep(stepName: string, duration: number): Promise<void> {
+    this.stepCount += 1;
+
+    if (this.skip || this.stepCount < this.nonPlanStepCount) return;
```

Comment:
I changed it as isSkipped, which doesn't really explain much :D

But I am planning to overhaul this logic because of parallel-sleep we discussed. Here is what I have in mind:

sleepFor/sleep/run will have an object with three callbacks:
- onSkip: when a previous step has executed and we shouldn't do anything (this won't really exist as a callback but as a condition)
- onExecute: when we run the step
- onCached: when the steps in the request has the result of this step (step has run before)

Reply:
Nicee, sounds great.